### PR TITLE
Misc. bug fixes from lgtm.com

### PR DESCRIPTION
**Pull request by esbena**
### Changes

This PR contains fixes for three independent bugs discovered by lgtm.com.

17fcf00 fixes an unfortunate typo in a regexp, see the commit message for a full description.

abfc666 and a682109 are trivial, and from the migrations folder, so I am not sure if you care for fixes there any more.

### Tests

`npm test` does not succeed on my machine (Ubuntu 16.04, node 8.4.0) for the develop branch.
This PR does not change these failures, or add any new failures. So I assume the PR should count as passing the tests -- I will let Travis decide...

For the record, here are the failures I observe:

- shared.ops.purchase successful purchase purchases food:
- shouldDo Every X Days should compute daily nextDue values:
- shouldDo Certain Days of the Week should compute weekly nextDue values:
- shouldDo Monthly - Every X Months on a specified date should compute monthly nextDue values:
- shouldDo Every X Years should compute yearly nextDue values:

### Addtional information

There are additional easily fixable alerts at https://lgtm.com/projects/g/HabitRPG/habitrpg/alerts/?mode=list, but I think a programmer more experienced with the habitica code base/style should take a look at those.
NB: I think the alerts about "Unused dependency" are false positives because the system does not recognize `.vue` files currently -- that will be fixed in a few weeks :).

(Full disclosure: I work on lgtm's JavaScript analysis.)

06f01285-dbbf-4fd4-8cb8-e98fb3a7a4b4

### CODE REVIEW

**Comment by esbena** [comment]
The Travis failures seem unrelated to my commits. Can you confirm?


```
1) Purchasing a group plan for group adds months to members with existing recurring subscription (Amazon):
AssertionError: expected 4.057975255749701 to be within 3..4
```
and
```
1a48f4dd/lib/internal-rules'
npm ERR! code EINTEGRITY
npm ERR! sha1-vj/4DKKkvYgbcqycLD3Vu3J2sqU= integrity checksum failed when using sha1: wanted sha1-vj/4DKKkvYgbcqycLD3Vu3J2sqU= but got sha1-6lRZU9lMrnRowoxc61rOePa1wac=. (24272 bytes)
```

**Comment by Alys** [comment]
Oh good catch on that regex, thanks!

Some of the migrations files are reused either for the same purpose or as templates for creating new ones, so bug fixes are good (typically we wouldn't reuse the really old ones but fixes won't hurt IMO).

I've just checked the travis outputs and the two failures that I can see are not related to your code - we're getting them in all PRs at the moment:
- "NotAuthorized: This item is not currently available for purchase" - that's probably because candy is the current flavour of food, rather than the normal foods, although I'm just guessing and haven't investigated
- "AssertionError: expected 4.057975255749701 to be within 3..4" - I think our current understanding is that's just an end-of-month bug, which should be fixed in the tests but I believe it's not high priority

I can't find this error in the travis output: "npm ERR! sha1-vj/4DKKkvYgbcqycLD3Vu3J2sqU= integrity checksum failed when using sha1: wanted sha1-vj/4DKKkvYgbcqycLD3Vu3J2sqU= but got sha1-6lRZU9lMrnRowoxc61rOePa1wac=. (24272 bytes)"
Are you still seeing it anywhere? Can you give a link?

**Comment by esbena** [comment]
> I can't find this error in the travis output: "npm ... "
> Are you still seeing it anywhere? Can you give a link?

No, it is gone now. My browser history tells me I looked at https://travis-ci.org/HabitRPG/habitica/jobs/295076965, which  is the log with 

>  1) shared.ops.purchase successful purchase purchases food:
>     NotAuthorized: This item is not currently available for purchase.

Could Travis have realized it was an internal thing that merited a retry?

This is my current Travis view:
image

Yesterday, the second red cross was a red exclamation mark, and the log file for that row led to the pasted `npm ERR ...`.

**Comment by Alys** [comment]
@esbena All the tests are passing now. There were some problems with the tests over the past few days that weren't related to your changes.

**Comment by esbena** [comment]
Thank you for checking up on that.

File: migrations/20151021_usernames_emails_lowercase.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ function findUsers(gt){
       console.log('User: ', countUsers, user._id);
 
       var update = {
-        $set: {};
+        $set: {}
```
Comment (paglias) [inline migrations/20151021_usernames_emails_lowercase.js:42]: this should be a comma not a semicolon that's not allowed in objects

File: website/server/models/user/schema.js [hunk 2]
```diff
@@ -251,7 +251,7 @@ let schema = new Schema({
     gear: {
       owned: _.transform(shared.content.gear.flat, (m, v) => {
         m[v.key] = {type: Boolean};
-        if (v.key.match(/[armor|head|shield]_warrior_0/) || v.gearSet === 'glasses') {
+        if (v.key.match(/(armor|head|shield)_warrior_0/) || v.gearSet === 'glasses') {
```
Comment (paglias) [inline website/server/models/user/schema.js:254]: @SabreCat does this look good to you?

File: migrations/20151021_usernames_emails_lowercase.js (hunk 1 above)
Comment (esbena) [reply migrations/20151021_usernames_emails_lowercase.js:42]: This PR *removes* the errorneous semicolon.

`$set` is the only property in this object literal, so replacing it with a comma would be another mistake:
```
var update = {
    $set: {},
};
```

Comment (paglias) [reply migrations/20151021_usernames_emails_lowercase.js:42]: @esbena oh sorry, I completely mixed the added and remove code :)

**Review by SabreCat — APPROVED** [review]

**Comment by SabreCat** [comment]
Thank you for these, @esbena, and welcome to the ranks of the Blacksmiths! You're well on your way to tier 2 in fact, with a much-appreciated stack of fixes across two PRs.

**Pull request merged**